api/src/Security/Http/AccessToken/Oidc/OidcDiscoveryTokenHandler.php

```diff
@@ -0,0 +1,98 @@
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Component\Security\Core\Exception\BadCredentialsException;
+use Symfony\Component\Security\Http\AccessToken\AccessTokenHandlerInterface;
+use Symfony\Component\Security\Http\AccessToken\Oidc\Exception\MissingClaimException;
+use Symfony\Component\Security\Http\AccessToken\Oidc\OidcTokenHandler;
+use Symfony\Component\Security\Http\AccessToken\Oidc\OidcTrait;
+use Symfony\Component\Security\Http\Authenticator\FallbackUserLoader;
+use Symfony\Component\Security\Http\Authenticator\Passport\Badge\UserBadge;
+use Symfony\Contracts\Cache\CacheInterface;
+use Symfony\Contracts\Cache\ItemInterface;
+
+/**
+ * Completes {@see OidcTokenHandler} with OIDC Discovery and configuration stored in cache.
+ */
+final class OidcDiscoveryTokenHandler implements AccessTokenHandlerInterface
+{
+    use OidcTrait;
+
+    public function __construct(
+        #[Autowire('@cache.app')]
+        private CacheInterface $cache,
+        #[Autowire('@jose.jws_loader.oidc')]
+        private JWSLoader $jwsLoader,
+        #[Autowire('%env(OIDC_SERVER_URL)%')]
+        private string $issuer,
+        private string $claim = 'email',
+        private int $ttl = 3600,
+        private ?LoggerInterface $logger = null,
+    ) {
+    }
+
+    public function getUserBadgeFrom(string $accessToken): UserBadge
+    {
+        if (!class_exists(JWSVerifier::class) || !class_exists(Checker\HeaderCheckerManager::class)) {
+            throw new \LogicException('You cannot use the "oidc_discovery" token handler since "web-token/jwt-signature" and "web-token/jwt-checker" are not installed. Try running "composer require web-token/jwt-signature web-token/jwt-checker".');
+        }
+
+        if (!class_exists(HttpClient::class)) {
+            throw new \LogicException('You cannot use the "oidc_discovery" token handler since "symfony/http-client" is not installed. Try running "composer require symfony/http-client".');
+        }
+
+        $oidcConfiguration = json_decode($this->cache->get('oidc.configuration', function (ItemInterface $item): string {
+            $item->expiresAfter($this->ttl);
+            $response = HttpClient::create()->request('GET', rtrim($this->issuer, '/') . '/.well-known/openid-configuration');
+
+            return $response->getContent();
+        }), true, 512, \JSON_THROW_ON_ERROR);
+
+        $keyset = JWKSet::createFromJson(
+            $this->cache->get('oidc.jwkSet', function (ItemInterface $item) use ($oidcConfiguration): string {
+                $item->expiresAfter($this->ttl);
+                $client = HttpClient::create();
+                $response = $client->request('GET', $oidcConfiguration['jwks_uri']);
+                // we only need signature key
+                $keys = array_filter($response->toArray()['keys'], static fn (array $key) => 'sig' === $key['use']);
+
+                return json_encode(['keys' => $keys]);
+            })
+        );
```

Comment:
It doesn't look easy but we should probably try/catch a bunch of errors no? I'm thinking about the `toArray()`(what happens if `jwks_uri` is wrong or unreacheable? Above, the `JSON_THROW_ON_ERROR` isn't catched. (I started reviewing this so maybe that the tokenhandler exceptions are handled in another file, please ignore if that's the case)